controls/V-72049.rb

```diff
@@ -45,19 +51,51 @@
 environment variables."
   tag "fix_id": "F-78401r1_fix"
 
-  # @todo - test for values more restrictive than 077
-  file_lines = command('grep -i -s umask /home/*/.*').stdout.split("\n")
-  if !file_lines.nil? and !file_lines.empty?
-    file_lines.each do |curr_line|
-      file_name = curr_line.split(':').first
-      describe command("grep -i umask #{file_name}") do
-        its('stdout.strip') { should match %r{^umask\s+.*077} }
-      end
+  # Get all interactive users
+  ignore_shells = non_interactive_shells.join('|')
+  
+  # Get home directory for users with UID >= 1000 or UID == 0 and support interactive logins.                                                              
+  findings = Set[]
+  dotfiles = Set[]
+  umasks = {}
+  interactive_users = users.where{ !shell.match(ignore_shells) && (uid >= 1000 || uid == 0)}.entries
+
+  # For each user, build and execute a find command that identifies initialization files                                                                   
+  # in a user's home directory.                                                                                                                            
+  interactive_users.each do |u|
+    # Get user's initialization files
+    dotfiles = dotfiles + command("find #{u.home} -xdev -maxdepth 2 ( -name '.*' ! -name '.bash_history' ) -type f").stdout.split("\n")
```

Comment:
This is going to tank NFS systems.
We probably need a safety valve that skips users that aren't part of a local filesystem.

Reply:
Doesn't -xdev prevent against this?

Reply:
@djhaynes Not really. The issue is if you point at something that's an NFS mount (which `/home/<user>` often is) it will delve into the NFS mount but not break out of it.

You may be able to use the `df -l` command to detect whether or not the target is a local filesystem and then skip with a warning.

Reply:
I took a pass at updating this control to skip NFS filesystems. Let me know what you think.

Reply:
@djhaynes I would still try to use `df -l` if you can and only run down the ones that are local.

I just tried it on my local system and `df -l` returns `1` if the filesystem is not local.

This will ensure that you not only ignore NFS but also CIFS, AFS, SSHFS, etc...

Reply:
@trevor-vaughan I updated the control to use the df -l command to check if the filesystem was local. This PR should be all set now.